vNext/Parallel-Tests.ps1

```diff
@@ -21,16 +39,28 @@ $ExecuteCmd =
 
 foreach ($a in $assemblies)
 {
+    $running = @(Get-Job | Where-Object { $_.State -eq 'Running' })
+    if ($running.Count -ge $maxDegreeOfParallelism) {
+        $running | Wait-Job -Any | Out-Null
+    }
+
+    if (Receive-CompletedJobs) { $failed = $true }
+	
     $xmlName = 'xUnit-Results-' + [System.IO.Path]::GetFileNameWithoutExtension($a) + '.xml'
     $outXml = $(Join-Path $outDir $xmlName)
     $cmdLine = $xunitRunner.FullName + ' ' + $a + ' ' + $testFilter + ' -xml ' + $outXml + ' -parallel none -noshadow -verbose' 
     Write-Host $cmdLine
-    Start-Job $ExecuteCmd -ArgumentList $cmdLine -Name $([System.IO.Path]::GetFileNameWithoutExtension($a))
+    Start-Job $ExecuteCmd -ArgumentList $cmdLine -Name $([System.IO.Path]::GetFileNameWithoutExtension($a)) | Out-Null
+    Write-Host ''
 }
 
-Get-Job | Wait-Job | Receive-Job
+# Wait for all jobs to complete and results ready to be received
+Wait-Job * | Out-Null
 
-if (Get-Job -State Failed)
+if (Receive-CompletedJobs) { $failed = $true }
```

Comment:
what do you mean? Not too familiar with PowerShell, so that's the easiest way I could make it work, but I'm open to suggestions if you'll bear with me